request-handler/src/main/java/au/org/aodn/aws/wps/RequestParserFactory.java

```diff
@@ -3,36 +3,43 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-
+import au.org.aodn.aws.wps.exception.InvalidRequestException;
 import java.util.Map;
 
 /**
  * Created by craigj on 8/08/17.
  */
-public class RequestParserFactory {
+public class RequestParserFactory  {
     Logger logger = LoggerFactory.getLogger(RequestParserFactory.class);
 
-    public RequestParser getRequestParser(AwsApiRequest request) {
+    public RequestParser getRequestParser(AwsApiRequest request) throws InvalidRequestException {
         if (request.getHttpMethod().equals("POST")) {
             return new XmlBodyParser(request.getBody());
-        } else if (request.getHttpMethod().equals("GET")) {
+        }
+        else if (request.getHttpMethod().equals("GET")) {
             return new QueryStringParameterParser(request);
-        } else {
-            // TODO: handle as required by spec
-            Map<String, String> parameters = request.getQueryStringParameters();
+        }
+        else {
             Map<String, String> headers = request.getHeaders();
 
+            try  {
+                // TODO: handle as required by spec
+                Map<String, String> parameters = request.getQueryStringParameters();
+                for (String key : parameters.keySet()) {
+                    logger.error("  - Query parameter: Key [" + key + "], Value [" + parameters.get(key) + "]");
+                }
+            }
+            catch (Exception e){
+                throw new InvalidRequestException(e);
+            }
+
             //  Log some debugging information
             logger.error("Unsupported HTTP method invoked: " + request.getHttpMethod());
 
-            for(String key : headers.keySet()) {
+            for (String key : headers.keySet()) {
                 logger.error("  - Query header: Key [" + key + "], Value [" + headers.get(key) + "]");
             }
 
-            for(String key : parameters.keySet()) {
```

Comment:
You could just null check the parameters object here - so that we log any parameters if they exist.